main.go

```diff
@@ -5,14 +5,26 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"path/filepath"
+	"strings"
 	"time"
 
 	"github.com/foundriesio/fioconfig/internal"
 	"github.com/urfave/cli/v2"
 )
 
 func NewApp(c *cli.Context) (*internal.App, error) {
-	return internal.NewApp(c.String("config"), c.String("secrets-dir"), c.Bool("unsafe-handlers"), false)
+	app, err := internal.NewApp(c.String("config"), c.String("secrets-dir"), c.Bool("unsafe-handlers"), false)
+	if err != nil {
+		return nil, err
+	}
+	if c.Command.Name == "renew-cert" {
+		return app, nil
+	}
+	stateFile := filepath.Join(c.String("config"), "cert-rotation.state")
+	handler := internal.NewCertRotationHandler(app, stateFile, "")
```

Comment:
I think the `<EST Server>` CLI argument should be passed as the third param here.

Reply:
for some reason I am getting the following.
For some reason it keeps sending request to `https://localhost:8443` regardless of the CLI param.

```
bin/fioconfig-linux-amd64 -c /work/fio/prjs/lmp-device-register/device renew-cert  foobar
foobar
FOO: foobar
2022/11/03 18:11:56 Performing certificate renewal
2022/11/03 18:11:56 Executing step: Generate new certificate
https://localhost:8443
2022/11/03 18:11:56 Unable to submit certificate signing request: Post "https://localhost:8443/simplereenroll": x509: certificate is valid for msul-ota, not localhost

```

Reply:
i hit this yesterday. I'm leaning towards this is a bug and not a feature. But it also highlights the need for a little more logging. I've got a 2nd issue with HSMs i'm looking at and there's not enough logging for me understand the real issue.

Reply:
fixed in the force-push. The function was doing too much, so i created two new function `NewCertRotationHandler` which requires the EstServer and `RestoreCertRotationHandler` which is designed to resume a rotation after a failure.

THis change caused enough changes that it kind of made me need to force-push changes so that it could be done sanely.

Reply:
fixed in force-push